VMM_Implementaion.md

```diff
@@ -0,0 +1,107 @@
+# [VMM Impl](../src/vmm/src/lib.rs)
+
+The `Vmm` is the main structure that holds all the VMM resources. It is responsible for the high level management of the VMM. It is also responsible for the communication with the API and the VMM action handlers.
+
+```rs
+pub struct Vmm {
+    vm: KvmVm<WrappedExitHandler>,
+    kernel_cfg: KernelConfig,
+    guest_memory: GuestMemoryMmap,
+    device_mgr: Arc<Mutex<IoManager>>,
+    event_mgr: EventManager<Arc<Mutex<dyn MutEventSubscriber + Send>>>,
+    exit_handler: WrappedExitHandler,
+    #[cfg(target_arch = "aarch64")]
+    num_vcpus: u64,
+}
+
+```
+- `vm` - The KVM VM object. It is created when the VMM starts. It is wrapped in a `KvmVm` object that provides a safe interface to the KVM API. The `KvmVm` object also holds a reference to the `exit_handler` object. The `exit_handler` is responsible for handling the VM exits. The `KvmVm` object is responsible for running the VM. 
+- `kernel_cfg` - The kernel configuration. It holds the kernel image, the kernel command line and the kernel start address.
+- `guest_memory` - The guest memory object. It is created when the VMM starts. It is wrapped in a `GuestMemoryMmap` object that provides a safe interface to the guest memory.
+- `device_mgr` - The device manager object. It is created when the VMM starts. It is responsible for managing the devices attached to the VMM. 
+- `event_mgr` - The event manager object. It is created when the VMM starts. It is responsible for managing the events that are generated by the devices attached to the VMM. 
+- `exit_handler` - The exit handler object. It is created when the VMM starts. It is responsible for handling the VM exits. 
```

Comment:
lot of redundancy in these statements. You can just say
`exit_handler`: responsible for handling VM exits.